---
title: "To do list"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: pdf_document
---

```{r setup, include=FALSE}
library(tidyverse)
knitr::opts_chunk$set(echo = TRUE)
```

## To do


### Reading 

* Learn about `spatialExperiment` 
  * Go through [this textbook](https://lmweber.org/OSTA-book/spatialexperiment.html)
  * Chapter 8
  * Read [bioarxiv paper](https://www.biorxiv.org/content/10.1101/2021.01.27.428431v2)
  * Check out workshops from last bioconductor conference [here](https://lmweber.org/OSTAWorkshopBioc2021/), [here](https://drighelli.github.io/SpatialExperiment_Bioc2021/index.html)


### MI to ST conversion

* Make Vectra compatible with `spatialExperiment`
  * Determine slot/structure for adding clinical data
  * Determine what goes in `rowData` slot. Potentially create a lookup table for types of popular markers and :
    * whether they are surface/nucleour/phenotypic/membrane
    * what types of cells they tend to mark
  * Write function for image data
    * Each pixel is a row?
    * If segmentation is defined, write function for extracting cells and putting into table format
    
* Data 
  * TNBC MIBI data
  * Put Simon's mIF data in `data` folder
    * Do brief exploratory analysis
    * Add this data to datasets document
    * Work towards making this compatible with Spatial Experiment

 
    
### Data package   

* Explore `ExperimentData` packages on Bioconductor
  * Check out [ST example data](https://bioconductor.org/packages/devel/data/experiment/html/STexampleData.html)
  * Check out [JP's](https://bioconductor.org/packages/3.13/data/experiment/html/ARRmData.html)
  * Check out [Sean Davis's](https://bioconductor.org/packages/3.13/data/experiment/html/ChimpHumanBrainData.html)
  * Check out [Stephanie's Visium package](https://bioconductor.org/packages/3.13/data/experiment/html/TENxPBMCData.html)
  * Think about which dataset you would want to turn into a package
  * Go through [guidelines](https://bioconductor.org/packages/release/bioc/vignettes/AnnotationHub/inst/doc/CreateAHubPackage.html) on how to create ExperimentHub package

### Other

* CRAN task view on single cell imaging data
  * http://bioconductor.org/packages/release/bioc/html/pRoloc.html
  * https://cran.r-project.org/web/packages/spatialTIME/index.html

## Recently completed

* Learn about `spatialExperiment` 
  * Go through [this textbook](https://lmweber.org/OSTA-book/spatialexperiment.html)
  * Chapter 1
  * Chapter 2
  * Chapter 3
  * Chapter 4
  * Chapter 5
  * Chapter 6
  * Chapter 7
 
* Make Vectra compatible with `spatialExperiment`
  * Check out `read10xVisium()` function, which creates a SpatialExperiment object from the raw input files expected from the 10x Genomics processing software ([link](https://github.com/drighelli/SpatialExperiment/blob/master/R/read10xVisium.R))
  * Write function for tabular data
  
    
## Ideas

* `MultiplexUtils` package to store the multiplex conversion functions?
* Call multiplex data package MIexampleData

## Ongoing questions

* Types of Bioconductor packages include: analysis software packages, annotation packages, data packages, workflow packages, online books
  * What are each of these?
  * Is `spatialExperiment` a workflow package?
* When creating a data package, would it be best to have data from one platform (just Vectra) or from multiple platforms?
  * Contains Visium and seqFISH data
* What software do you use to make the pretty figures?
* Why is `spatialCoords` its own slot rather than being a `colData` variable? is this because it was built on top of `singleCellExperiment`? Does having its own slot add functionality?
* Did you (Lukas) right the whole spatial transcriptomics book yourself?
  * Including the preprocessing steps section?
* What is the purpose of the `DropletUtils` package? What is Droplet data?

### About spatial transcriptomics

* Can `VistoSeg` be used to count/summarize the pixels in a cell?
* What is a molecule?
* How many genes are there typically in an ST experiment?
* From a data structure perspective, what are the differences between spatial transcriptomics and multiplex single cell imaging?
  * ST doesn't have single cell masks, right?
  * ST is wayyy more multiplex in the sense that there are hundreds if not thousands of genes
  * ST has counts and MI has continuous marker intensities
    * Are there "intensities" at all for ST?

### About read10xVisium.R

* Is it ok for the image data argument to be empty?
* Does it all the types of files at once?
  * Can you explain the expected file structure a little bit?

### Angie questions

* What is the file structure for typical Vectra data after Inform processing? 
* How challenging is it to get the .tiff images?
* How are file names / ids set up for the Vectra instrument?
* What image preprocessing occurs before Inform?
* Data dictionary somewhere?
  * What is the "Confidence" column for Vectra data?
  * What are the TMA Sector/Row/Column/Field variables?
  * Are column names consistent for each analysis?
  
